articles/azure-app-configuration/enable-dynamic-configuration-aspnet-core.md

```diff
@@ -281,10 +272,11 @@ A *sentinel key* is a special key used to signal when configuration has changed.
     ```
     ---
     
-    The middleware uses the refresh configuration specified in the `AddAzureAppConfiguration` method in `Program.cs` to trigger a refresh for each request received by the ASP.NET Core web app. For each request, a refresh operation is triggered and the client library checks if the cached value for the registered configuration setting has expired. If it's expired, it's refreshed.
-
     > [!NOTE]
-    > To ensure the configuration is refreshed, add the middleware as early as appropriate to your request pipeline so it will not be short-circuited by another middleware in your application.
+    > The middleware monitors the sentinel key or any other keys you registered for refreshing in the `ConfigureRefresh` call in the previous step. It is triggered upon every incoming request to your application. However, the middleware will only issue requests to check the value in App Configuration when the refresh time you set has passed. When a change is detected, it will either update all the configuration if the sentinel key is used or update the registered keys' value only. Note:
```

Comment:
>refresh time [](http://example.com/codeflow?start=311&length=12)

"refresh time" could be replaced with "cache expiration time" to maintain consistency with the `SetCacheExpiration` method name

Reply:
I was trying to avoid "cache expiration" but I guess it's inevitable. Updated as you suggested.